src/grblogtools/api.py

```diff
@@ -60,26 +65,34 @@ def progress(self, section="nodelog") -> dict:
         )
 
     def common_log_data(self):
-        """Extract summary data to be joined to progress logs."""
-
-        # The summary data is joined to progress log on LogFilePath + LogNumber.
-        # This method separately calls summary() whenever a progress log is created,
-        # so perhaps it should be cached. The cache would need to be invalidated
-        # if parse() is ever called again.
-        common_columns = [
-            "LogFilePath",
-            "LogNumber",
-            "Log",
-            "ModelFilePath",
-            "ModelFile",
-            "Model",
-            "Seed",
-            "Version",
-        ]
-        summary = self.summary()
-        return summary.loc[:, summary.columns.isin(common_columns)]
-
-    def summary(self):
+        """Extract common data to be joined to progress and summary dataframes.
+        This could be cached in future and invalidated by .parse()"""
+        common = pd.DataFrame(
+            [
+                {
+                    "LogFilePath": logfile,
+                    "LogNumber": lognumber,
+                    "ModelFilePath": parser.header_parser.get_summary().get(
+                        "ModelFilePath"
+                    ),
+                    "Seed": parser.header_parser.get_parameters().get("Seed", 0),
+                    "Version": parser.header_parser.get_summary().get("Version"),
+                }
+                for logfile, lognumber, parser in self.parsers
+            ]
+        )
+        common = common.dropna(axis="columns", how="all")
+        if "ModelFilePath" in common:
+            common = common.assign(
+                ModelFile=lambda df: df["ModelFilePath"].apply(
+                    lambda p: None if p is None else Path(p).parts[-1].partition(".")[0]
```

Comment:
It could be none if there is no "Read model from MPS file ... " or similar in one of the logs, e.g. when a model is formulated and solved directly in an API instead of through gurobi_cl. The lines above drop the entire column if _all_ ModelFilePaths are None, but if we have some nulls and some not, the null check is still needed I think.